(tree) Added op bunching processing to shared tree (#23732)
microsoft/FluidFramework: .changeset/purple-ideas-switch.md

---
"@fluidframework/tree": minor
---
---
"section": deprecation
---

> The SharedTreeCore.processCore on SharedTreeCore is now deprecated

Reviewer: ```suggestion
The SharedTreeCore.processCore function on SharedTreeCore is now deprecated
```

> A new function `processMessagesCore` has been added in its place which will be called to process multiple messages instead of a single one on the channel. This is part of a feature called "Op bunching" where contiguous ops in a grouped batch are bunched and processed together by the shared object.

Reviewer 2: ```suggestion
A new function `processMessagesCore` has been added in place of `processCore`, which will be called to process multiple messages instead of a single one on the channel. This is part of a feature called "Op bunching" where contiguous ops in a grouped batch are bunched and processed together by the shared object.
```

> A new function `processMessagesCore` has been added in its place which will be called to process multiple messages instead of a single one on the channel. This is part of a feature called "Op bunching" where contiguous ops in a grouped batch are bunched and sent together to the shared object for processing.

Reviewer 3: ```suggestion
A new function `processMessagesCore` has been added in its place which will be called to process multiple messages instead of a single one on the channel. This is part of a feature called "Op bunching" where contiguous ops in a grouped batch are bunched and ingested together by shared object for processing.
```
I'm not married to the phrasing I'm suggesting. I mostly want to avoid the word "sent" because I fear it might cause confusion since "sending" is generally used for over-the-network exchanges.
